application-forum-ui/src/main/resources/ForumCode/Actions.xml

```diff
@@ -171,10 +181,13 @@ ${spaceName}.${newAnswerName}
           #set($discard = $conversationDoc.delete())
           ## if we are a topic delete the answers
           #if($topicObj)
-           #foreach($item in $services.query.hql(", BaseObject as obj where doc.fullName=obj.name and obj.className='ForumCode.AnswerClass' and doc.parent='${conversationDoc.fullName}'").execute())
-            #set($itemdoc = $xwiki.getDocument($item))
-            #set($ok = $itemdoc.deleteWithProgrammingRights())
-           #end           
+            ## Depending on XE version, we consider parent-child relationship for Pre NS or space-subspace for NS.
+            #set($hierarchyCondition = 'doc.parent=:parent or doc.space=:space')
```

Comment:
When Answers will become subspaces, the binding values will be (for NS):
`doc.parent=Forums.MyForum.MyTopic.WebHome`
`doc.space=Forums.MyForum.MyTopic`
This should work fine. Am I missing something? 
Thanks!